src/manifolds/Tucker.jl

```diff
@@ -0,0 +1,876 @@
+
+@doc raw"""
+Tucker{N, R, D, 𝔽} <: AbstractManifold{𝔽}
+
+The manifold of $N_1 \times \dots \times N_D$ real-valued or complex-valued tensors of
+fixed multilinear rank $(R_1, \dots, R_D)$ . If $R_1 = \dots = R_D = 1$, this is the
+manifold of rank-1 tensors.
+
+# Representation in HOSVD format
+
+Any tensor $\mathcal{A}$ on the Tucker manifold can be represented in HOSVD
+[^DeLathauwer2000] form
+```math
+\mathcal{A} = (U_1,\dots,U_D) \cdot \mathcal{C}
+```
+where $\mathcal C \in \mathbb{F}^{R_1 \times \dots \times R_D}$ and, for $d=1,\dots,D$,
+the matrix $U_d \in \mathbb{F}^{N_d \times R_d}$ contains the singular vectors of the
+$d$th unfolding of $\mathcal{A}$
+
+# Tangent space
+
+The tangent space to the Tucker manifold at
+$\mathcal{A} = (U_1,\dots,U_D) \cdot \mathcal{C}$ is [^Koch2010]
+```math
+T_{\mathcal{A}} \mathcal{M} =
+\bigl\{
+(U_1,\dots,U_D) \cdot \dot{\mathcal{C}}
++ \sum_{d=1}^D \bigl(
+    (U_1, \dots, U_{d-1}, \dot{U}_d, U_{d+1}, \dots, U_D)
+    \cdot \mathcal{C}
+\bigr)
+\bigr\}
+```
+where $\dot{\mathcal{C}}$ is arbitrary, $\dot{U}_d^{\mathrm{H}}$ is the Hermitian adjoint of
+$U_d$, and $\dot{U}_d^{\mathrm{H}} U_d = 0$ for all $d$.
+
+# Constructor
+    Tucker(n⃗::NTuple{D, Int}, r⃗::NTuple{D, Int}[, field = ℝ])
+
+Generate the manifold of $N_1 \times \dots \times N_D$ tensors of fixed multilinear rank
+$(R_1, \dots, R_D)$
+
+[^DeLathauwer2000]:
+    > Lieven De Lathauwer, Bart De Moor, Joos Vandewalle: "A multilinear singular value decomposition"
+    > SIAM Journal on Matrix Analysis and Applications, 21(4), pp. 1253-1278, 2000
+    > doi: [10.1137/S0895479896305696](https://doi.org/10.1137/S0895479896305696)
+
+[^Koch2010]:
+    > Othmar Koch, Christian Lubic, "Dynamical Tensor approximation"
+    > SIAM Journal on Matrix Analysis and Applications, 31(5), pp. 2360-2375, 2010
+    > doi: [10.1137/09076578X](https://doi.org/10.1137/09076578X)
+"""
+struct Tucker{N,R,D,𝔽} <: AbstractManifold{𝔽} end
+function Tucker(n⃗::NTuple{D,Int}, r⃗::NTuple{D,Int}, field::AbstractNumbers=ℝ) where {D}
+    @assert is_valid_mlrank(n⃗, r⃗)
+    return Tucker{n⃗,r⃗,D,field}()
+end
+
+#=
+HOSVD{T, D}
+
+Higher-order singular value decomposition of an order D tensor with eltype T
+fields:
+* U: singular vectors of the unfoldings
+* core: core tensor
+* σ : singular values of the unfoldings
+=#
+struct HOSVD{T,D}
+    U::NTuple{D,Matrix{T}}
+    core::Array{T,D}
+    σ::NTuple{D,Vector{T}}
+end
+
+"""
+TuckerPoint{T, D}
+
+An order D tensor of fixed multilinear rank and entries of type T. The tensor is
+represented in HOSVD form. See also [`Tucker`](@ref).
+
+# Constructors:
+    TuckerPoint(core::AbstractArray{T, D}, factors::Vararg{MtxT, D}) where {T, D, MtxT <: AbstractMatrix{T}}
+
+A tensor of the form (factors[1], …, factors[D]) ⋅ core
```

Comment:
Should "(factors[1], …, factors[D])" maybe be in backtracks?
Would it be reasonable to explain also what T and D are?